optimum/habana/accelerate/data_loader.py

```diff
@@ -1,18 +1,269 @@
+import math
 from typing import Callable, List, Optional, Union
 
 import torch
 from accelerate.data_loader import (
     _PYTORCH_DATALOADER_KWARGS,
     BatchSamplerShard,
-    DataLoaderDispatcher,
     DataLoaderShard,
+    DataLoaderStateMixin,
     IterableDatasetShard,
     SeedableRandomSampler,
 )
-from accelerate.utils import RNGType
+from accelerate.state import GradientState
+from accelerate.utils import (
+    RNGType,
+    concatenate,
+    find_batch_size,
+    get_data_structure,
+    is_torch_version,
+    send_to_device,
+    slice_tensors,
+)
+from accelerate.utils.operations import _gpu_broadcast, is_tensor_information, recursively_apply
 from torch.utils.data import BatchSampler, DataLoader, IterableDataset
 
-from .state import GaudiAcceleratorState
+from .state import GaudiAcceleratorState, GaudiPartialState
+from .utils import GaudiDistributedType
+
+
+def initialize_tensors(data_structure):
+    """
+    Recursively initializes tensors from a nested list/tuple/dictionary of [`~utils.TensorInformation`].
+
+    Returns:
+        The same data structure as `data` with tensors instead of [`~utils.TensorInformation`].
+    """
+
+    def _initialize_tensor(tensor_info):
+        return torch.zeros(*tensor_info.shape, dtype=tensor_info.dtype)
```

Comment:
I see that the latest release of Accelerate uses `torch.empty`: https://github.com/huggingface/accelerate/blob/v0.25.0/src/accelerate/utils/operations.py#L220
Any specific reason to use `torch.zeros` here?

Reply:
yes. meeting error like "Error when trying to cast Long to Int, Input values range [0, 72340172838010880] exceeds Int range [-2147483648, 2147483647]" if use torch.empty